### PR TITLE
Export metrics to api: schedulingTime and runningTime

### PR DESCRIPTION
**Pull request by yohengyang**
## Description
schedulingTime and runningTime are helping for performance troubleshooting. Export them to /ui/api/query/queryId json.

### CODE REVIEW

**Comment by yohengyang** [comment]
> Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **yangyuheng**. This is most likely caused by a git client misconfiguration; please make sure to:
> 
> 1. check if your git client is configured with an email to sign commits `git config --list | grep email`
> 2. If not, set it up using `git config --global user.email email@example.com`
> 3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

File: core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java [hunk 1]
```diff
@@ -262,6 +262,8 @@ private static QueryStats immediateFailureQueryStats()
                 new Duration(0, MILLISECONDS),
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:262]: commit message is too long

Is this two changes in single commit? Could we split them?

File: core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java [hunk 2]
```diff
@@ -262,6 +262,8 @@ private static QueryStats immediateFailureQueryStats()
...
                 new Duration(0, MILLISECONDS),
                 new Duration(0, MILLISECONDS),
+                new Duration(0, MILLISECONDS),
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:265]: How `runningTime` is different than `io.trino.execution.QueryStateTimer#executionTime`?
Are the metrics already in `io.trino.execution.QueryStateTimer`?

**Review by sopel39 — COMMENTED** [review]
please rebase instead of using merge branches

File: core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java (hunk 2 above)
Comment (yohengyang) [reply core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:265]: https://github.com/trinodb/trino/blob/51d91df5db92b4d51346f6b13b649b7568a2c972/core/trino-main/src/main/java/io/trino/event/QueryMonitor.java#L586

File: core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java (hunk 1 above)
Comment (yohengyang) [reply core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:262]: only metric changes which can not split more.

Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:262]: commit message is still too long

File: core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java (hunk 2 above)
Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:265]: Code in `io.trino.event.QueryMonitor#logQueryTimeline(io.trino.execution.QueryInfo)` should use metrics computed here.

Why `max` is used there?

Comment (yohengyang) [reply core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java:265]: Yep, there is no need compute metrics dulplicated both in io.trino.event.QueryMonitor and  io.trino.execution.QueryStateMachine. Is that better remoe compute  logic from QueryMonitor

**Comment by mosabua** [comment]
I'm closing this pull request due to inactivity. Feel free to reopen after remaining to continue.

**Pull request closed without merging**